Makefile

```diff
@@ -1,137 +1,205 @@
+	$(DOCKER_RUN_73) ./tests/e2e_tests $(INFECTION)
 
-test-infection-xdebug-73: build-xdebug-73
-	$(DOCKER_RUN_73) php bin/infection --threads=4
+.PHONY: test-infection-phpdbg
+test-infection-phpdbg:		## Runs Infection with PHPDBG against itself
+test-infection-phpdbg: test-infection-phpdbg-72 test-infection-phpdbg-73
 
-.PHONY: test-e2e-xdebug test-e2e-xdebug-72 test-e2e-xdebug-73
-#e2e tests with xdebug
-test-e2e-xdebug: test-e2e-xdebug-72 test-e2e-xdebug-73
+.PHONY: test-infection-phpdbg-72
+test-infection-phpdbg-72:	## Runs Infection with PHPDBG against itself in PHP 7.2
+test-infection-phpdbg-72: $(DOCKER_RUN_72_IMAGE)
+	# TODO: calculate the number of thread here instead of using the arbitrary number 4
+	# TODO: check other occurrences
+	$(DOCKER_RUN_72) phpdbg -qrr bin/infection --threads=4
 
-test-e2e-xdebug-72: build-xdebug-72 $(INFECTION)
-	$(DOCKER_RUN_72) ./tests/e2e_tests $(INFECTION)
+.PHONY: test-infection-phpdbg-73
+test-infection-phpdbg-73:	## Runs Infection with PHPDBG against itself in PHP 7.3
+test-infection-phpdbg-73: $(DOCKER_RUN_73_IMAGE)
+	$(DOCKER_RUN_73) phpdbg -qrr bin/infection --threads=4
 
-test-e2e-xdebug-73: build-xdebug-73 $(INFECTION)
-	$(DOCKER_RUN_73) ./tests/e2e_tests $(INFECTION)
+.PHONY: test-infection-xdebug
+test-infection-xdebug:		## Runs Infection with xdebug against itself
+test-infection-xdebug: test-infection-xdebug-72 test-infection-xdebug-73
 
-.PHONY: build-xdebug-72 build-xdebug-73
-#Building images with xdebug
+.PHONY: test-infection-xdebug-72
+test-infection-xdebug-72:	## Runs Infection with xdebug against itself in PHP 7.2
+test-infection-xdebug-72: $(DOCKER_RUN_72_IMAGE)
+	$(DOCKER_RUN_72) ./bin/infection --threads=4
 
-build-xdebug-72: vendor devTools/Dockerfile-php72-xdebug.json
-devTools/Dockerfile-php72-xdebug.json: devTools/Dockerfile-php72-xdebug
-	docker build -t infection_php72 -f devTools/Dockerfile-php72-xdebug .
-	docker image inspect infection_php72 >> devTools/Dockerfile-php72-xdebug.json
+.PHONY: test-infection-xdebug-73
+test-infection-xdebug-73:	## Runs Infection with xdebug against itself in PHP 7.3
+test-infection-xdebug-73: $(DOCKER_RUN_73_IMAGE)
+	$(DOCKER_RUN_73) ./bin/infection --threads=4
 
-build-xdebug-73: vendor devTools/Dockerfile-php73-xdebug.json
-devTools/Dockerfile-php73-xdebug.json: devTools/Dockerfile-php73-xdebug
-	docker build -t infection_php73 -f devTools/Dockerfile-php73-xdebug .
-	docker image inspect infection_php73 >> devTools/Dockerfile-php73-xdebug.json
 
-#style checks/ static analysis
-.PHONY: analyze cs-fix cs-check phpstan validate auto-review
-analyze: cs-check phpstan validate
+#
+# Rules from files (non-phony targets)
+#---------------------------------------------------------------------------
 
-# PHP-CS-Fixer is checked by PrettyCI
-.PHONY: analyze-ci
-analyze-ci: phpstan validate
+# TODO: some tools such as Box & PHP-CS-Fixer could be handled in a better way to ensure we are using the last version.
```

Comment:
Noteworthy TODO. This as of now is _already_ an issue at the very least locally. Indeed if you run:

```
make compile
```

it will download Box, but if you update the Box URL, the existing one won't be invalidated... This is however a problem that should be tackled in a dedicated PR IMO

Reply:
I have thought about this issue before, for a future PR: how about moving all phars into a folder (maybe `/build`?), and having a `make clean` command which deletes all files in the folder.

Reply:
I used `.tools` in this PR but could be `build` as well it's not an issue. I liked to use `.tools` since it makes it cacheable...

Maybe an alternative would be combine the PHAR with a cache key with the cache key dependent on the corresponding URL: if the URL changes so does the cache key hence the PHAR needs to be downloaded again

Reply:
`.tools` works as well, i just use `build` in most of my personal projects, and didn't notice `the `.tools` in this PR 😅 

> Maybe an alternative would be combine the PHAR with a cache key with the cache key dependent on the corresponding URL: if the URL changes so does the cache key hence the PHAR needs to be downloaded again

That would be a solution i think.

Reply:
For another PR though as I really didn't want to think of the details tonight :)